### PR TITLE
03 get api topics

### PR DESCRIPTION
**Pull request by seb-the-architect**
please please please accept my pull request

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+const request = require("supertest");
+const app = require(`../app.js`);
+const testData = require("../db/data/test-data/index.js")
+//const pool = require("../db/connection.js")
+const seed = require("../db/seeds/seed.js")
+
+beforeEach(() => {return seed(testData)})
+
+//afterAll(() => {return pool.end()})
```
Comment (seb-the-architect) [inline __tests__/app.test.js:9]: forgot to uncomment line 4 and 9 :(((

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        .then((response) => 
+        {
+            expect(response.body.topics.length === 3);
+            for(let eachObject of response.body.topics)
+            {
+                expect.objectContaining({slug:expect.any(String), description:expect.any(String)});
+            }
+        });
+    });
+});
```
Comment (elijaharley) [inline __tests__/app.test.js:27]: Nice work, this test is working really well now.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const app = express();
+
+/*
+GET /api/topics
+-------------------
+Responds with: an array of topic objects, each of which should have the following properties:
+slug
+description
+-------------------
+*/
```
Comment (elijaharley) [inline app.js:13]: Don't forget to remove comments after you've finished using them.

File: controller/sql-query-controller.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const model = require("../model/sql-query-model.js");
+
+exports.getAllTopics = async function(req, res)
+{
+    const allTopics = await model.queryAllTopics();
+    
+    try { res.status(200).send({"topics": allTopics}) } 
+    catch (error) { console.error(error); }
+}
```
Comment (elijaharley) [inline controller/sql-query-controller.js:9]: Really nice setup of the controller, Seb, and good implementation of the try catch syntax for async/await.

File: model/sql-query-model.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+//Database: nc-news
+//Table: topics
+
+//Returns all topics as a list of objects
+exports.queryAllTopics = async function()
+{
+    const allTopics = await pool.query("SELECT * FROM topics");
+    console.log(allTopics.rows);
+    return allTopics.rows;
+}
```
Comment (elijaharley) [inline model/sql-query-model.js:12]: Don't forget to remove console.logs and comments.

**Review by elijaharley — APPROVED** [review]
Really good work on this, Seb. Don't forget to remove your console.logs and comments once you've finished with them.

**Comment by seb-the-architect** [comment]
Updated and merged successfully.

**Pull request closed without merging**